src/GovUk.Education.ExploreEducationStatistics.Data.Model.Tests/Repository/FootnoteRepositoryTests.cs

```diff
@@ -6,12 +6,2253 @@
+                    Release = release,
+                    Subject = subject1
+                }
+            };
+
+            var contextId = Guid.NewGuid().ToString();
+
+            await using (var context = InMemoryStatisticsDbContext(contextId))
+            {
+                await context.Subject.AddAsync(subject1);
+                await context.Filter.AddAsync(filter1);
+                await context.IndicatorGroup.AddAsync(indicatorGroup1);
+                await context.Release.AddAsync(release);
+                await context.ReleaseSubject.AddRangeAsync(releaseSubjects);
+                await context.SaveChangesAsync();
+            }
+
+            await using (var context = InMemoryStatisticsDbContext(contextId))
+            {
+                var repository = BuildFootnoteRepository(context);
+
+                //  Get the footnotes applying to filter item 1 and indicator 1
+                var results = await repository.GetFilteredFootnotes(
+                    releaseId: release.Id,
+                    subjectId: subject1.Id,
+                    filterItemIds: ListOf(filter1.FilterGroups[0].FilterItems[0].Id),
+                    indicatorIds: ListOf(indicatorGroup1.Indicators[0].Id));
+
+                // Check that only footnotes applying to filter item 1, or indicator 1
+                // are returned.
+
+                // Any footnotes applying to filter item 2 or indicator 2 should be excluded.
+                // This includes the footnotes for filter 1 and indicator 2, filter group 1 and indicator 2,
+                // filter item 1 and indicator 2, filter item 2 and indicator 1, filter item 2 and indicator 2.
+                Assert.Equal(7, results.Count);
+
+                Assert.Equal(releaseFootnotes[0].FootnoteId, results[0].Id);
+                Assert.Equal("Applies to filter 1", results[0].Content);
+
+                Assert.Equal(releaseFootnotes[1].FootnoteId, results[1].Id);
+                Assert.Equal("Applies to filter 1 group 1", results[1].Content);
+
+                Assert.Equal(releaseFootnotes[2].FootnoteId, results[2].Id);
+                Assert.Equal("Applies to filter item 1", results[2].Content);
+
+                Assert.Equal(releaseFootnotes[4].FootnoteId, results[3].Id);
+                Assert.Equal("Applies to indicator 1", results[3].Content);
+
+                Assert.Equal(releaseFootnotes[6].FootnoteId, results[4].Id);
+                Assert.Equal("Applies to filter 1 and indicator 1", results[4].Content);
+
+                Assert.Equal(releaseFootnotes[8].FootnoteId, results[5].Id);
+                Assert.Equal("Applies to filter 1 group 1 and indicator 1", results[5].Content);
+
+                Assert.Equal(releaseFootnotes[10].FootnoteId, results[6].Id);
+                Assert.Equal("Applies to filter item 1 and indicator 1", results[6].Content);
+            }
+        }
+
+        [Fact]
+        public async Task GetFilteredFootnotes_FilterByFilters()
```

Comment:
Should this be called "GetFilteredFootnotes_FilterByFilterItems"?  Only because from the test setup it seems that we're passing `filterItemIds` as the main filtering parameter.

Reply:
Yes good spot, I think you are right. I've renamed it, thanks! 🙂